config.js

```diff
@@ -247,6 +249,30 @@ function setRankedUnrankedOptionsDefaults(rankedUnrankedOptions, argv) {
     }
 }
 
+function getValidFilename(filename) {
+    // convert any other character than letters (A-Z a-z), numbers (0-9), hypens (-), underscore (_)
+    // to a hyphen (-)
+    return filename.replace(/[^\w]/g, "-");
+}
+
+function exportLogfileFilename(argvLogfile) {
+    if (argvLogfile === "") {
+        // provide default filename based on start date and time
+        //
+        const filenameDate = `gtp2ogs-logfile-${exports.start_date.toISOString()}`;
+        const validFilenameDate = getValidFilename(filenameDate);
```

Comment:
These 2 lines are duplicate, you can rewrite this as
```Js
function exportLogfileFilename(argvLogfile) {
        if (argvLogfile === "") {
            // provide default filename based on start date and time
            //
            argvLogfile = `gtp2ogs-logfile-${exports.start_date.toISOString()}`;
        }
        const validFilename = getValidFilename(argvLogfile);
        if (argvLogfile !== validFilename) {
            console.log (`Invalid logfile name "${argvLogfile}" has been automatically renamed`
                         + ` to "${validFilename}".\nValid logfile name can only be composed of`
                         + ` letters (A-Z a-z), numbers (0-9), hyphens (-), underscores (_).\n`);
        }
        exports.logfile = validFilename;
    }
}
```

Reply:
yes, except that we modify our parameter here, would it be better to create a new const for that?

also, console.log message should be slightly adjusted (just saying that logfile has been renamed, no need to say it's invalid)
but yes this is better like this 🎉

Reply:
i made the no existing name check to a function is it good this way? (we reuse getValidFilename twice, is it clear that getLogfileFilename gets the initial logfile filename ?)

tested on beta to work, 

![fixed logfilename](https://user-images.githubusercontent.com/38690718/88060309-5c11e200-cb66-11ea-940e-427d679ce896.png)

and also updated #332 and #332 's unit testing to match these changes
i find this code a lot cleaner than what i was :)